packages/@uppy/transloadit/src/index.ts

```diff
@@ -1009,27 +874,16 @@ export default class Transloadit<
       })
     })
 
-    const watcher = this.assemblyWatchers[uploadID]
-    return watcher.promise.then(() => {
+    return this.watcher.promise.then(() => {
       closeSocketConnections()
-
-      const assemblies = assemblyIDs.map((id) => this.getAssembly(id))
-
-      // Remove the Assembly ID list for this upload,
-      // it's no longer going to be used anywhere.
-      const uploadsAssemblies = { ...this.getPluginState().uploadsAssemblies }
-      delete uploadsAssemblies[uploadID]
-      this.setPluginState({ uploadsAssemblies })
-
       this.uppy.addResultData(uploadID, {
-        transloadit: assemblies,
+        transloadit: [this.assembly.status],
       })
     })
   }
 
-  #closeAssemblyIfExists = (assemblyID?: string) => {
-    if (!assemblyID) return
-    this.activeAssemblies[assemblyID]?.close()
+  #closeAssemblyIfExists = () => {
+    this.assembly?.close()
   }
 
   #onError = (err: { name: string; message: string; details?: string }) => {
```

Comment:
now that there is only one assembly, we could re-enable this todo code (see todo below)
```suggestion
  #onError = (err: { name: string; message: string; details?: string }) => {
      this.#closeAssemblyIfExists()
```